src/ctl/ctl-sink.c

```diff
@@ -87,50 +89,46 @@ static void sink_handle_get_parameter(struct ctl_sink *s,
 	/* wfd_audio_codecs */
-	if (rtsp_message_read(m, "{<>}", "wfd_audio_codecs") >= 0) {
-		r = rtsp_message_append(rep, "{&}",
-					"wfd_audio_codecs: AAC 00000007 00");
-		if (r < 0)
-			return cli_vERR(r);
-	}
+//	check_and_response_option("wfd_audio_codecs", "wfd_audio_codecs: AAC 00000007 00");
+	check_and_response_option("wfd_audio_codecs", "LPCM 00000003 00, AAC 0000000f 00, AC3 00000007 00");
 	/* wfd_client_rtp_ports */
-	if (rtsp_message_read(m, "{<>}", "wfd_client_rtp_ports") >= 0) {
-		char wfd_client_rtp_ports[128];
-		sprintf(wfd_client_rtp_ports,
-					"wfd_client_rtp_ports: RTP/AVP/UDP;unicast %d 0 mode=play", rstp_port);
-		r = rtsp_message_append(rep, "{&}",
-					wfd_client_rtp_ports);
-		if (r < 0)
-			return cli_vERR(r);
-	}
+	char wfd_client_rtp_ports[128];
+	sprintf(wfd_client_rtp_ports, "RTP/AVP/UDP;unicast %d 0 mode=play", rstp_port);
+	check_and_response_option("wfd_client_rtp_ports", wfd_client_rtp_ports);
+	check_and_response_option("wfd_content_protection", "none");
+	check_and_response_option("wfd_display_edid", "0001 00ffffffffffff0051f38f50010000000e100104a51d10ff2f0000a057499b2610484f000000010101010101010101010101010101011a36809c70381f403020350025a510000018000000fc00496e7465726e616c204c43440a000000fd003c3c9a9a0e00000000000000000000000000000000000000000000000000000030");
+	check_and_response_option("wfd_connector_type", "05");
+	check_and_response_option("microsoft_cursor", "none");
+	check_and_response_option("microsoft_rtcp_capability", "none");
+	check_and_response_option("wfd_idr_request_capability", "1");
+	check_and_response_option("microsoft_latency_management_capability", "none");
+	check_and_response_option("microsoft_format_change_capability", "none");
+	check_and_response_option("microsoft_diagnostics_capability", "none");
+	check_and_response_option("intel_friendly_name", "miraclecast");
+	check_and_response_option("intel_sink_manufacturer_name", "GNU Linux");
+	check_and_response_option("intel_sink_model_name", "Arch linux");
+	check_and_response_option("intel_sink_device_URL", "http://github.com/albfan/miraclecast");
 
 	/* wfd_uibc_capability */
-	if (rtsp_message_read(m, "{<>}", "wfd_uibc_capability") >= 0 && uibc) {
-		char wfd_uibc_capability[512];
-		sprintf(wfd_uibc_capability,
-			"wfd_uibc_capability: input_category_list=GENERIC;"
-         "generic_cap_list=Mouse,SingleTouch;"
-         "hidc_cap_list=none;port=none");
-		r = rtsp_message_append(rep, "{&}", wfd_uibc_capability);
-		if (r < 0)
-			return cli_vERR(r);
+	if (uibc) {
+//		check_and_response_option("wfd_uibc_capability",
+//								  "input_category_list=GENERIC;"
+//				                  "generic_cap_list=Mouse,SingleTouch;"
+//				                  "hidc_cap_list=none;"
+//				                  "port=none");
+		check_and_response_option("wfd_uibc_capability",
```

Comment:
process this output depending on sink. machine used was a laptop (that's why it has Multitouch) but miraclecast has only GENERIC implemented